miden-crypto/src/merkle/merkle_tree.rs

```diff
@@ -166,6 +156,18 @@ impl MerkleTree {
     }
 }
 
+impl IntoIterator for MerkleTree {
+    type Item = InnerNodeInfo;
+    type IntoIter = InnerNodeIterator;
+
+    fn into_iter(self) -> Self::IntoIter {
+        InnerNodeIterator {
+            nodes: self.nodes,
+            index: 1, // index 0 is just padding, start at 1
+        }
+    }
+}
```

Comment:
@adr1anh @plafer LMK if this is along the lines you were thinking. And whether you would want me to do the same for other `From<&T>` impls in the repo.

Reply:
As mentioned in #444, if this PR doesn't make sense then we should just close it (and perhaps document the learnings for when we eventually forget and someone brings this up again 🙂).